WordSearch.java

```diff
@@ -0,0 +1,113 @@
+import java.util.*;
+
+/**
+ * Created by Alice Colt.
+ */
+public class WordSearch {
+
+
+    // Search for a WORD - using list
+
+    static List<String> wordSearchGrid(char[][] characters,
+                                       int row,
+                                       int column,
+                                       Dictionary dictionary){
+
+
+        // List for keeping all the solutions
+        List<String> solution = new LinkedList<>();
+
+        boolean[][] visited = new boolean[row][column];
+
+
+        for (int index = 0; index < row; index++)
+        {
+            for (int indexj = 0; indexj < column; indexj++)
+            {
+                Set<String> validWord = wordGetter(Character.toString(characters[index][indexj]),
+                                                  index, indexj,
+                                                  row, column,
+                                                  characters,visited,dictionary);
+
+                // Add to solution list
+                solution.addAll(validWord);
+            }//for
+        }//for
+
+
+        // Return solution list
+        return new ArrayList<>(solution);
```

Comment:
I don't think you need to convert to a list - I think the assignment even says "return a set of words", so I think it would be fine just to return the set.